### PR TITLE
Added documentation examples for `unhashable-dict-key`.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Refs https://github.com/PyCQA/pylint/issues/5953

### CODE REVIEW

**Comment by mbyrnepr2** [comment]
Note perhaps this checker should be expanded to cover the following TypeError:
```python
{[1, 2, 3]: 4}
```

File: doc/data/messages/u/unhashable-dict-key/bad.py [hunk 1]
```diff
@@ -0,0 +1 @@
+{1: 2}[[1]]  # [unhashable-dict-key]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/unhashable-dict-key/bad.py:1]:
```suggestion
fruits_number = {"apple": 42}
print(f"Number of apples : {fruits_number[['apple']]}")  # [unhashable-dict-key]
```

File: doc/data/messages/u/unhashable-dict-key/good.py [hunk 2]
```diff
@@ -1 +1 @@
-# This is a placeholder for correct code for this message.
+{1: 2}[1]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/unhashable-dict-key/good.py:1]:
```suggestion
fruits_number = {"apple": 42}
print(f"Number of apples : {fruits_number['apple']}")
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I had a really hard time to understand the example with so much abstraction 😄

**Comment by Pierre-Sassoulas** [comment]
>  perhaps this checker should be expanded to cover the following TypeError

I think this is a really neat proposal !

File: doc/data/messages/u/unhashable-dict-key/bad.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+fruits_number = {"apple": 42}
+print(f"Number of apples : {fruits_number[['apple']]}")  # [unhashable-dict-key]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/u/unhashable-dict-key/bad.py:2]:
```suggestion
print(f"Number of apples : {{'apple': 42}[['apple']]}")  # [unhashable-dict-key]
```
The message might only be raised when the dict is defined in place like this ?

Comment (mbyrnepr2) [reply doc/data/messages/u/unhashable-dict-key/bad.py]: Looks like it yeah. I've updated the examples but no worries if you would like to change them again - I am not sure the more I think about them :D

**Review by Pierre-Sassoulas — APPROVED** [review]
Let's not overthink it 😄 I wonder how often this message is raised though, we're really careful by not using any inference on the content of the dict.

**Pull request merged**